Extensions/xkit_patches.js

```diff
@@ -915,6 +915,208 @@ XKit.extensions.xkit_patches = new Object({
 						tags: post.tags.join(","),
 					};
 				},
+				/**
+				 * Get the posts on the screen without the given tag
+				 * @param {String} without_tag - Class that the posts should not have
+				 * @param {Boolean} can_edit - Whether the posts must be editable
+				 * @return {Array<Object>} The posts
+				 */
+				get_posts: async function(without_tag, can_edit) {
+					var posts = [];
+
+					var selector = "[data-id]";
+
+					var editLabel = await XKit.interface.translate("Edit");
+
+					await XKit.css_map.getCssMap();
+					var recommended_selector = XKit.css_map.keyToCss("recommendationReason");
+
+					var selection = $(selector);
+					selection = selection.not("." + without_tag);
+
+					selection.each(function() {
+						var $this = $(this);
+
+						// recommended post
+						var recommended = $this.find(recommended_selector);
+						if (recommended.length > 0) {
+							return;
+						}
```

Comment:
i don't think extensions should be wholesale blind to recommended posts, they're still posts
for quick tags specifically i'm pretty sure this would also prevent activating on your own pinned posts in peepr